controllers/chat_controller.py

```diff
@@ -0,0 +1,49 @@
+import discord
+from discord import Interaction
+from config import Config
+
+# Sub role ids
+Tier1 = int(Config.CONFIG["Discord"]["Tier1RoleID"])
+Tier2 = int(Config.CONFIG["Discord"]["Tier2RoleID"])
+Tier3 = int(Config.CONFIG["Discord"]["Tier3RoleID"])
+GiftedTier1 = int(Config.CONFIG["Discord"]["GiftedTier1RoleID"])
+GiftedTier2 = int(Config.CONFIG["Discord"]["GiftedTier2RoleID"])
+GiftedTier3 = int(Config.CONFIG["Discord"]["GiftedTier3RoleID"])
+sub_roles = {Tier1, Tier2, Tier3, GiftedTier1, GiftedTier2, GiftedTier3}
+
+chat_modes = {
+    "t3_only": {
+        "allowed_roles": {Tier3, GiftedTier3},
+        "slowmode": 0,
+    },
+    "normal": {
+        "allowed_roles": sub_roles,
+        "slowmode": 15,
+    },
+    "t3_jail": {
+        "allowed_roles": sub_roles - {Tier3, GiftedTier3},
```

Comment:
t3 jail needs to allow _everyone_ but T3s. This would only allow subs that aren't T3.